### PR TITLE
zenith: remove .Sync from module default main.go in go sdk

### PR DESCRIPTION
**Pull request by sipsma**
Calling sync isn't required there and just creates confusion, so it's better to remove it.

---

Discord discussion: https://discord.com/channels/707636530424053791/1120503349599543376/1166774396820672512

### CODE REVIEW

**Review by helderco — COMMENTED** [review]
Without `Sync` you also don't need `context.Context`.

**Comment by sipsma** [comment]
> Without `Sync` you also don't need `context.Context`.

Yeah, also don't need `error`, I left them in since I wanted it to still be immediately obvious that it is valid to use context and error though.

Maybe I'll just add one more method that actually uses `context` and `error` and drop those from the other one, so then we quick examples of both?

**Comment by sipsma** [comment]
I also added comments to the example in the previous update, but my comments ended in a `"`, which triggered a very silly corner case where our gql schemas ended in `""""` instead of `"""` and broke 😄

The fix was trivial so I just pushed it here: https://github.com/dagger/dagger/pull/5963/commits/bcd701f7e0f3b53468f75fe0bd5d2aa8491b1f1a

File: core/integration/module_test.go [hunk 1]
```diff
@@ -359,14 +359,19 @@ func TestModuleGit(t *testing.T) {
...
+					With(daggerQuery(`{bare{containerEcho(stringArg:"hello"){stdout}}}`)).
+					Stdout(ctx)
+				require.NoError(t, err)
+				require.JSONEq(t, `{"bare":{"containerEcho":{"stdout":"hello\n"}}}`, out)
+			} else {
+				out, err := modGen.
+					With(daggerQuery(`{bare{myFunction(stringArg:"hello"){stdout}}}`)).
+					Stdout(ctx)
+				require.NoError(t, err)
+				require.JSONEq(t, `{"bare":{"myFunction":{"stdout":"hello\n"}}}`, out)
```
Comment (helderco) [inline core/integration/module_test.go:372]: I'll update the Python default to match Go. I did that added the tests to have unified test case but it's also nice for the user to compare the different implementations of the same thing.

**Pull request merged**